internal/sql/sql.go

```diff
@@ -155,6 +156,11 @@ func (s *DB) Transaction(fn func(Tx) error) error {
 		if !locked {
 			break
 		}
+		select {
```

Comment:
We should pass the context to `s.transaction` so we can use `db.BeginTx` (that accepts a context). It also feels a bit odd to pass a context into a method that has a (long() sleep in it. If we rely on the `context.Cause` higher up we should wrap it again because `BeginTx` doesn't return it.